pandas/_libs/window/indexers.pyx

```diff
@@ -143,4 +147,6 @@ def calculate_variable_window_bounds(
             # right endpoint is open
             if not right_closed and not center:
                 end[i] -= 1
-    return start, end
+    ref = (None if step is None or step == 1
```

Comment:
What is ref? And based on the typing this shouldn't be None (maybe rather an empty array?)

Reply:
`ref` is an array of locations for which the window bounds have been computed, and is `None` if all locations. For example, if `step=2` and no group-by was used then `ref=np.arange(0, num_values, step, dtype=np.int64)`. If group-by was used, the stepping is applied to each group separately, which leads to an irregular `ref`.

I could try to change from `None` to an empty array and see if everything works.

Reply:
Note that `ref` is needed for correctly slicing the index, especially when group-by was used. It is passed to `_slice_index`.

Reply:
I think to keep this feature very simple to start, let's just implement this for rolling with an integer offset (all other varieties should raise a `NotImplementedError`)

Reply:
> Can't the index be sliced with self.step? Why is another array needed?

The index cannot be sliced using `step` alone when group-by was used because then the `step` slicing should be applied to each group separately.

> I think to keep this feature very simple to start, let's just implement this for rolling with an integer offset (all other varieties should raise a NotImplementedError)

Presumably you mean integer `step`, right? For backward compatibility, `step=None` is equivalent to `step=1`. What currently supported cases should raise a `NotImplementedError`?

Reply:
Correct, an `integer (non-negative) step`

> What currently supported cases should raise a NotImplementedError?

* expanding
* ewm
* rolling(window="time-like offset")
* groupby.rolling/expanding/ewm